test/libsolidity/semanticTests/abiEncoderV2/struct/struct_short.sol

```diff
@@ -0,0 +1,14 @@
+pragma experimental ABIEncoderV2;
+
+contract C {
+    struct S { int a; uint b; bytes16 c; }
+    function f(S memory s) public pure returns (S memory q) {
+        q = s;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// f((int256,uint256,bytes16)): 0xff010, 0xff0002, "abcd" -> 0xff010, 0xff0002, "abcd"
+// f((int256,uint256,bytes16)): 0xff010, 0xff0002, 0x000000 -> 0xff010, 0xff0002, 0
+// f((int256,uint256,bytes16)): 0xff010, 0xff0002, 0x000000 -> 0x0ff010, 0xff0002, 0
```

Comment:
This call reverts in https://github.com/ethereum/solidity/pull/10192/commits/d1244018c4b5dac5c401b08b0b0d7913763750f0?file-filters%5B%5D=.cpp#diff-8257b09e03faebbd251d9cc2583d2ba4ddcc7b16a9edadd31bcce3c02f17eb46L492

Reply:
I guess it has to do with the actual length of the bytes array: the previous one was supposed to enforce 32, whereas this one has 31

Reply:
Yes, will need to change both of them to enforce specific lengths

Reply:
I am not sure how to encode this same way

Reply:
Also not sure. Regardless how many zeroes you use, it uses 32 bytes. Maybe @chriseth or @ekpyron know?

Reply:
yes, always 32 bytes. If you have trouble with the encoding, you can also write a single solidity function that calls multiple others.

Reply:
ah, sorry, adding it back now